### DISCUSSION TITLE
Dropping Windows 32bit legacy release

### DISCUSSION THREAD

**Discussion by Safihre** [sabnzbd/sabnzbd#2970] [General]
With Python 3.8 becoming end of life last month, I think it's time to stop shipping the 32bit/legacy Windows release.
Python packages we depend on are already dropping Python 3.8 support, causing me to have to make workarounds just for the 32bit/legacy release.
The download stats are already extremely low for the legacy release compared to the regular one: https://sabnzbd.org/download-stats

With that I would also like to remove the 32bit executables we have in the repo for unrar and par2.

Opening an Discussion in case I'm overlooking some big problems with this.

**Comment by thezoggy** [comment]
I would deff like to see us just get rid of 32bit util/legacy stuff. but prob makes sense for us to do that with something like 4.5? so we can at least have a clear break. and those that really want to rock the legacy os can just use older sab that should* still work until it doesnt.

**Reply by Safihre** [reply]
Why not 4.4.0? It's still in beta?

The download stats are just sooo low.
@jeffgeorge

**Reply by jcfp** [reply]
4.4 would be fine too, IMHO. As long as the version bump doesn't suggest only some minor bugfix/maintenance release.

**Reply by thezoggy** [reply]
Personally I'd like to see 4.4 out for them just because there are actual bug fixes (stalling) that people run into... But of course it's hard to know how many actual users would be really impacted. I guess worst case we could direct them to beta build that would still work and have the fixes

**Reply by Safihre** [reply]
@thezoggy Oke that seems reasonable.

**Comment by jeffgeorge** [comment]
I had a thought that we could check self-test stats to get an idea of how many 32bit installs are out there but whoops we strip all of that data by sending a custom user agent of just `SABnzbd/4.3.3`, so that's unfortunately a complete non-starter.

This level of deprecation feels like something that would warrant a 4.5.0 or 5.0.0 version bump. I agree it _probably_ impacts very very very few users out there, but still feels like something to save for a major version bump.

**Comment by Safihre** [comment]
Every week another dependency drops support for 3.8 and I have to make a workaround to keep it going.
https://github.com/sabnzbd/sabnzbd/pull/2974

So we would also be dropping Python 3.8 support.. @jcfp is that maybe a problem for Ubuntu? 
Otherwise I could just keep applying the workarounds.

**Reply by thezoggy** [reply]
last python 3.8 release came out only a few months ago:
[Python 3.8.20 - Sept. 6, 2024](https://www.python.org/downloads/release/python-3820/)

ubuntu 20.04 LTS (focal) has python 3.8.2
ubuntu 22.04 LTS (jammy) has python 3.10.6

debian 10.13 (buster) comes with python 3.7.3
debian 11.11 (bullseye) comes with python 3.9.2

doesnt synology still use python 3.8?

**Reply by Safihre** [reply]
True, but since October it's EOL, so that was the last one.
https://devguide.python.org/versions/
For the SAB code itself it's not a problem to support it, only its depencies that keep dropping it with updates. 

Synology uses 3.11 but will soon update to 3.13. They skipped 3.12.

**Reply by jcfp** [reply]
Debian:
```
 python3-defaults | 3.7.3-1  | buster   | source
 python3-defaults | 3.9.2-3  | bullseye | source
 python3-defaults | 3.11.2-1 | bookworm | source
 python3-defaults | 3.12.6-1 | trixie   | source
 python3-defaults | 3.12.7-1 | sid      | source
```

Ubuntu:
```
 python3-defaults | 3.6.5-3          | bionic          | source
 python3-defaults | 3.8.2-0ubuntu2   | focal           | source
 python3-defaults | 3.10.4-0ubuntu2  | jammy           | source
 python3-defaults | 3.12.3-0ubuntu1  | noble           | source
 python3-defaults | 3.12.6-0ubuntu1  | oracular        | source
 python3-defaults | 3.12.6-0ubuntu1  | plucky          | source
```

Their respective developement releases (trixie, plucky) are currently in the process of migrating to 3.13.

The only release affected by dropping 3.8 would be focal/20.04, for which the regular support window runs until April 2025. Supporting the release only requires that sab's own codebase supports Python 3.8, and any (new) dependencies have at least some version that works with both.

**Reply by thezoggy** [reply]
small typo - focal not jammy

**Comment by Safihre** [comment]
I dropped it now for 4.5. 
So 4.4 still supports 32bit/legacy, but it will show a warning to the user.

**Reply by sanderjo** [reply]
... same for python 3.8? Supported with 4.4, and to be dropped with 4.5? I think dropping support for python 3.8 should be in the Release Notes. Not sure about a warning from the SABnzbd code.

**Reply by Safihre** [reply]
No, 3.8 we will keep!

**Reply by thezoggy** [reply]
> I dropped it now for 4.5. So 4.4 still supports 32bit/legacy, but it will show a warning to the user.

coo, i know there is new 7zip 24.09  but nothing really that we need in it and rar 7.10 b1 (hoping that will be out of beta by the time were ready to do sab 4.5) and we can update all our external utilities to latest...  i assume we will still keep multipar around for a few more cycles until we have confidence that par2cmdline-turbo on windows is solid ?

**Reply by Safihre** [reply]
4.4 is in release candidate now, so not updating anything anymore.
Maybe for 4.4.1.

Indeed keeping Multipar for a bit, probably 4.6 removal? Not in a rush.

**Comment by thezoggy** [comment]
While updating my windows arm laptop (finally getting win11 24h2 update), I see that 32bit arm UWP being killed with it:

![image](https://github.com/user-attachments/assets/abca6c04-9da9-410d-9bf3-f235be4802f2)

ms does call out that its only arm not x86 just yet, but still figured I'd just note here as it gives even further incentive to just toss 32bit out 
https://www.microsoft.com/en-us/windows/windows-11-specifications#table3

**Reply by thezoggy** [reply]
- Win10 eol was yesterday, Windows 11 does not support 32-bit processors.

- Firefox 145 released yesterday, on linux dropped 32bit as well
